2019-02-06  Brendon Tetsuo Ng  <classvwt@lnxsrv07.seas.ucla.edu>

        * Assignment4/timespec.h (timespec_cmp): Use comparison operators to prevent
	overflow due to subtraction that occurred when dealing with files very long apart.
	
--- lib/timespec.h	2019-02-06 10:37:44.605108000 -0800
+++ lib/timespec.h	2019-02-06 10:38:44.883003000 -0800
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec < b.tv_sec)
+    return -1;
+  else if (a.tv_sec == b.tv_sec)
+    return 0;
+  else
+    return 1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
